### PR TITLE
how-to: update hardware-and-sofware-requirements

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Update hardware-and-sofware-requirements.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/3228
- Other reference link(s):

### CODE REVIEW

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ aliases: ['/docs/dev/how-to/deploy/hardware-recommendations/']
 
 ## About
```
Comment (yikeke) [inline hardware-and-software-requirements.md:11]:
```suggestion
```

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -26,9 +26,29 @@ As an open source distributed NewSQL database with high performance, TiDB can be
...
+### Control Machine
+
+| Software | Version |
+| :--- | :--- |
+| sshpass | 1.06 or later |
+| TiUP | 0.6.2 or later |
+
+> **Note:**
+>
+> [Deploy TiUP](/tiup/tiup-cluster.md) on the Control Machine to operate and manage TiDB clusters.
```
Comment (yikeke) [inline hardware-and-software-requirements.md:40]:
```suggestion
> It is required that you [deploy TiUP](/tiup/tiup-cluster.md) on the Control Machine to operate and manage TiDB clusters.
```

File: hardware-and-software-requirements.md [hunk 3]
```diff
@@ -26,9 +26,29 @@ As an open source distributed NewSQL database with high performance, TiDB can be
...
+
+| Software | Version |
+| :--- | :--- |
+| sshpass | 1.06 or later |
+| numa | 2.0.12 or later |
+
 ## Server recommendations
 
-You can deploy and run TiDB on the 64-bit generic hardware server platform in the Intel x86-64 architecture. The requirements and recommendations about server hardware configuration (ignoring the resources occupied by the operating system itself) for development, test, and production environments are as follows:
+You can deploy and run TiDB on the 64-bit generic hardware server platform in the Intel x86-64 architecture or the hardware server platform in the ARM architecture. The requirements and recommendations about server hardware configuration (ignoring the resources occupied by the operating system itself) for development, test, and production environments are as follows:
```
Comment (yikeke) [inline hardware-and-software-requirements.md:51]:
```suggestion
You can deploy and run TiDB on the 64-bit generic hardware server platform in the Intel x86-64 architecture or on the hardware server platform in the ARM architecture. The requirements and recommendations about server hardware configuration (ignoring the resources occupied by the operating system itself) for development, test, and production environments are as follows:
```

File: hardware-and-software-requirements.md [hunk 4]
```diff
@@ -44,6 +64,7 @@ You can deploy and run TiDB on the 64-bit generic hardware server platform in th
 > - In the test environment, the TiDB and PD instances can be deployed on the same server.
 > - For performance-related test, do not use low-performance storage and network hardware configuration, in order to guarantee the correctness of the test result.
 > - For the TiKV server, it is recommended to use NVMe SSDs to ensure faster reads and writes.
+> - If you only want to verify the features, follow [Quick Start Guide for TiDB](/quick-start-with-tidb.md) to deploy TiDB and test the features on a single machine.
```
Comment (yikeke) [inline hardware-and-software-requirements.md:67]:
```suggestion
> - If you only want to test and verify the features, follow [Quick Start Guide for TiDB](/quick-start-with-tidb.md) to deploy TiDB on a single machine.
```

File: hardware-and-software-requirements.md [hunk 5]
```diff
@@ -26,9 +24,29 @@ As an open source distributed NewSQL database with high performance, TiDB can be
...
+### Control Machine
+
+| Software | Version |
+| :--- | :--- |
+| sshpass | 1.06 or later |
+| TiUP | 0.6.2 or later |
+
+> **Note:**
+>
+> It is required that you [deploy TiUP](/tiup/tiup-cluster.md) on the Control Machine to operate and manage TiDB clusters.
```
Comment (yikeke) [inline hardware-and-software-requirements.md:38]:
```suggestion
> It is required that you [deploy TiUP](/production-deployment-using-tiup.md#第-2-步在中控机上安装-tiup-组件) on the Control Machine to operate and manage TiDB clusters.
```
Waiting on https://github.com/pingcap/docs-cn/pull/3228/files#r433695667

Comment (yikeke) [reply hardware-and-software-requirements.md:38]: Please see https://github.com/pingcap/docs-cn/pull/3228/files#r433772498 to fix the docs. @ran-huang

**Comment by ran-huang** [comment]
@yikeke Updated the link. PTAL again~

**Comment by yikeke** [comment]
> @yikeke Updated the link. PTAL again~

The linked doc is not ready yet. 😂 How about removing this link and merging this PR first? After we add the tiup docs, we add it back.

File: hardware-and-software-requirements.md [hunk 6]
```diff
@@ -26,9 +24,29 @@ As an open source distributed NewSQL database with high performance, TiDB can be
...
+### Control Machine
+
+| Software | Version |
+| :--- | :--- |
+| sshpass | 1.06 or later |
+| TiUP | 0.6.2 or later |
+
+> **Note:**
+>
+> It is required that you [deploy TiUP on the Control Machine](/tiup/tiup-documentation-guide.md) to operate and manage TiDB clusters.
```
Comment (ran-huang) [inline hardware-and-software-requirements.md:38]:
```suggestion
> It is required that you deploy TiUP on the Control Machine to operate and manage TiDB clusters.
```

**Pull request merged**